icu4c/source/i18n/unicode/messageformat2_formattable.h

```diff
@@ -64,6 +65,50 @@ namespace message2 {
         virtual ~FormattableObject();
     }; // class FormattableObject
 
+    /**
+     * The `DateInfo` struct represents all the information needed to
+     * format a date with a time zone. It includes an absolute date and a time zone name,
+     * as well as a calendar name. The calendar name is not currently used.
+     *
+     * @internal ICU 76 technology preview
+     * @deprecated This API is for technology preview only.
+     */
+    struct U_I18N_API DateInfo {
+        /**
+         * Milliseconds since Unix epoch
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UDate date;
+        /**
+         * IANA time zone name; "UTC" if UTC; empty string if value is floating
+         *
+         * @internal ICU 76 technology preview
+         * @deprecated This API is for technology preview only.
+         */
+        UnicodeString zoneName;
```

Comment:
should this be const?

Reply:
`DateInfo` is movable (because it can appear inline in `Formattable`, which is movable), so the members can't be const. I could change `DateInfo` to a class and make the members private, if you think that would be better.